machine_manager/models/machinery.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, fields, _
+
+
+class Machinery(models.Model):
+    _name = "machinery"
+    _description = "Manage your machines"
+
+    def _def_company(self):
+        return self.env.user.company_id.id
+
+    name = fields.Char('Machine Name', required=True)
+    company = fields.Many2one('res.company', 'Company', required=True,
+                              default=_def_company)
+    assetacc = fields.Many2one('account.account', string='Asset Account')
+    depracc = fields.Many2one('account.account', string='Depreciation Account')
+    year = fields.Char('Year')
+    model = fields.Char('Model')
+    product = fields.Many2one(
+        comodel_name='product.product', string='Associated product',
+        help="This product will contain information about the machine such as"
+        " the manufacturer.")
+    manufacturer = fields.Many2one(
+        comodel_name='res.partner', related='product.manufacturer',
+        readonly=True, help="Manufacturer is related to the associated product"
+        " defined for the machine.")
+    serial_char = fields.Char('Product Serial #')
+    serial = fields.Many2one('stock.production.lot', string='Product Serial #',
+                             domain="[('product_id', '=', product)]")
+    model_type = fields.Many2one('machine.model', 'Type')
+    status = fields.Selection([('active', 'Active'), ('inactive', 'Inactive'),
+                               ('outofservice', 'Out of Service')],
+                              'Status', required=True, default='active')
+    ownership = fields.Selection([('own', 'Own'), ('lease', 'Lease'),
+                                  ('rental', 'Rental')],
+                                 'Ownership', default='own', required=True)
+    bcyl = fields.Float('Base Cycles', digits=(16, 3),
```

Comment:
please avoid using cryptic abbreviations in field names. The days are long past where column names were limitted to 8 chars.